2-copy-of-code/lesson-16/tests/data/cartTest.js

```diff
@@ -136,4 +136,22 @@ describe('test suite: updateDeliveryOption', () => {
     expect(cart[0].deliveryOptionId).toEqual('1');
     expect(localStorage.setItem).toHaveBeenCalledTimes(0);
   });
+
+  it('does nothing if the delivery option does not exist', () => {
+    spyOn(localStorage, 'getItem').and.callFake(() => {
+      return JSON.stringify([{
+        productId: 'e43638ce-6aa0-4b85-b27f-e1d07eb678c6',
+        quantity: 1,
+        deliveryOptionId: '1'
+      }]);
+    });
+    loadFromStorage();
+
+    updateDeliveryOption('e43638ce-6aa0-4b85-b27f-e1d07eb678c6', 'does-not-exist');
+    expect(cart.length).toEqual(1);
+    expect(cart[0].productId).toEqual('e43638ce-6aa0-4b85-b27f-e1d07eb678c6');
+    expect(cart[0].quantity).toEqual(1);
+    expect(cart[0].deliveryOptionId).toEqual('1');
+    expect(localStorage.setItem).toHaveBeenCalledTimes(0);

```

Comment:
Thank you so much for your course. I learned a lot. Could you please help me here? Why do I get the following error on the last line? All other code is the same.

`expect(localStorage.setItem).toHaveBeenCalledTimes(0);
`
Error: <toHaveBeenCalledTimes> : Expected a spy, but got Function.
Usage: expect(<spyObj>).toHaveBeenCalledTimes(<Number>)
    at <Jasmine>
    at UserContext.<anonymous> (http://127.0.0.1:5500/tests/data/cartTest.js:139:34)
    at <Jasmine>